### PR TITLE
Introducing sentry, logging to console in dev and production settings

### PR DESCRIPTION
**Pull request by AriHrannar**
## Status
**Ready**

## Description
1. Production integration with Sentry
2. Most loggers now log in the console
3. Since cron runs in a separate daemon, it cant log to the console. Therefore we still log to files there, but for development purposes we mount them into a /logs directory in the root of the project.
4. Local and production settings
5. Runs well locally with docker compose AND deployable via docker cloud!

### CODE REVIEW

**Comment by AriHrannar** [comment]
@skabbi how much experience do you have with Docker, Docker compose and [Docker cloud](cloud.docker.com)?

**Comment by skabbi** [comment]
@AriHrannar Very limited, just know the basics. Why do you ask?

**Comment by AriHrannar** [comment]
@skabbi mostly wondering how much that interests you :) I can at least invite you to the docker cloud organization we have (just send me your email on Facebook or something)

About Sentry, it seems like the free version is limited to only 1 user in the cloud right now.
Options:
1. Setup on-premise, but that requires some coding (need to generate a key and put it as an environment variable)
2. Share my Account (think we should do this for now at least)

**Comment by Ragnar-H** [comment]
@AriHrannar the tests seem to be breaking, is that because there is a lot of infrastructure that's changed?

**Comment by AriHrannar** [comment]
Maybe the docker-compose.test.yml isnt 100% in sync with the new changes. Ill check this out either today or tomorrow

**Comment by AriHrannar** [comment]
@Ragnar-H is fixed now :)

**Comment by Ragnar-H** [comment]
> @Ragnar-H is fixed now :)

I'm fixed now? 
That's great 😁

**Comment by Ragnar-H** [comment]
> About Sentry, it seems like the free version is limited to only 1 user in the cloud right now.
Options:
Setup on-premise, but that requires some coding (need to generate a key and put it as an environment variable)
Share my Account (think we should do this for now at least)

How about creating a Veritus account?
Is that an option?

File: README.md [hunk 1]
```diff
@@ -58,3 +58,13 @@ It also works to use this during development, as docker-compose mounts the sourc
...
+
+## Sentry
+### Setup
+Run 
+```
+docker-compose run --rm web upgrade
+```
+to setup Sentry locally and creating a user. 
+
+Run ```docker-compose up``` to start all the services and go to localhost:9000 to see sentry.
```
Comment (Ragnar-H) [inline README.md]: Not blocking.
But remember you can do singular quotes (`) when the command is in-line

File: docker-compose.yml [hunk 2]
```diff
@@ -31,3 +31,59 @@ services:
     depends_on:
       db:
         condition: service_healthy
+  # Sentry setup
+  sentry:
+    image: sentry:8.17.0
+    environment:
+      # Run `docker-compose run web config generate-secret-key`
```
Comment (Ragnar-H) [inline docker-compose.yml]: Could you expand on this?
If I run this command to generate a secret key, why is it set in the line below?

File: docker-compose.yml [hunk 3]
```diff
@@ -19,10 +19,11 @@ services:
       - POSTGRES_PASSWORD=kaninei3i3j94jginerign
       - SECRET_KEY=nin3on3inri3nnwoge
       - DB_HOST=db
-      - DEBUG=false
+      - DEBUG='true' # Dont have this true in production!
```
Comment (Ragnar-H) [inline docker-compose.yml]: should this have quotes around it?

File: docker-compose.test.yml [hunk 4]
```diff
@@ -25,8 +25,7 @@ services:
       - ALLOWED_HOSTS=[]
       - CORS_ORIGIN_WHITELIST=()
       - GATHER_CASES_CRON_TIME_SECONDS=1440
-      - SENTRY_DSN=notNeededForDevelopment
-      - DJANGO_SETTINGS_MODULE=politech_backend.settings.production
+      - DJANGO_SETTINGS_MODULE=politech_backend.settings.local
```
Comment (Ragnar-H) [inline docker-compose.test.yml]: Nitpicking.
I prefer calling the environments `development, test, staging, production`

So `local = development` ?

File: docker-compose.test.yml [hunk 5]
```diff
@@ -16,7 +16,7 @@ services:
 
   sut:
     build: .
-    command: "bash -c 'python3 src/manage.py migrate && python3 src/manage.py loaddata init_data.json && python3 src/manage.py test'"
+    command: sh .entrypoint.sh
```
Comment (Ragnar-H) [inline docker-compose.test.yml]: \o/

File: docker-compose.yml [hunk 6]
```diff
@@ -21,11 +21,11 @@ services:
       - DB_HOST=db
       - ALLOWED_HOSTS=["*"]
       - CORS_ORIGIN_WHITELIST=('localhost:3000',)
-      - GATHER_CASES_CRON_TIME_SECONDS=1 #440
+      - GATHER_CASES_CRON_TIME_SECONDS=1440
       - DJANGO_SETTINGS_MODULE=politech_backend.settings.local
     volumes:
       - ./:/code
-    command: "bash -c 'python3 src/manage.py migrate && python3 src/manage.py loaddata init_data.json && cron && cd src && gunicorn -b 0.0.0.0:8000 politech_backend.wsgi'"
+    command: sh .entrypoint.sh
```
Comment (Ragnar-H) [inline docker-compose.yml]: ❤️ this

File: docker-compose.test.yml [hunk 7]
```diff
@@ -16,7 +16,7 @@ services:
 
   sut:
     build: .
-    command: sh entrypoint.sh
+    command: "bash -c 'python3 src/manage.py migrate && python3 src/manage.py loaddata init_data.json && python3 src/manage.py test'"
```
Comment (Ragnar-H) [inline docker-compose.test.yml]: :(

Comment (Ragnar-H) [inline docker-compose.test.yml]: I think you want a [MakeFile](https://www.gnu.org/software/make/manual/make.html) for this.

It gives you access to `Package.json`-esqe commands.
Nothing that needs to be implemented now. But to keep in mind for the future

**Comment by AriHrannar** [comment]
I have a "Veritus" organization on Sentry but it seems its limited to 1 "person account" in the free tier. As in, one email. I guess we could have a Veritus account, but I would prefer to get all emails to my email account  to follow the error reporting :)

I think that is fine for now, if you guys ever need access, ill send you the password

**Comment by AriHrannar** [comment]
@Ragnar-H did you maybe review an old version of the files? All your comments appear to be out of date

**Comment by Ragnar-H** [comment]
> @Ragnar-H did you maybe review an old version of the files? All your comments appear to be out of date

I review commit-by-commit. Once I started doing this, I got a much deeper understanding of what pull requests are about :)

This is one of the reasons I **love** rebasing master, and using Squash before merging ❤️

**Pull request merged**